code/datums/components/snowballs.dm

```diff
@@ -0,0 +1,30 @@
+/datum/component/snowballs
+	var/turf/source_turf
+
+/datum/component/snowballs/Initialize()
+	if(!isturf(parent))
+		return COMPONENT_INCOMPATIBLE
+	else
+		source_turf = parent
+	RegisterSignal(parent, list(COMSIG_ATTACKHAND), .proc/start_snowball)
+
+/datum/component/snowballs/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ATTACKHAND)
+	. = ..()
+
+/datum/component/snowballs/proc/start_snowball(turf/T, mob/user)
+	if(!ON_COOLDOWN(source_turf, "snowball", 6 SECONDS))
+		actions.start(new /datum/action/bar/icon/callback(user, T, rand(3 SECONDS, 5 SECONDS), /datum/component/snowballs/proc/form_snowball,
+		list(user), 'icons/misc/xmas.dmi', "snowball", null, null, src), user)
+		return TRUE
+
+/datum/component/snowballs/proc/form_snowball(mob/user)
+	user.visible_message("<b>[user]</b> gathers up some snow and rolls it into a snowball!", "You gather up some snow and roll it into a snowball!")
+	var/obj/item/reagent_containers/food/snacks/snowball/S = new /obj/item/reagent_containers/food/snacks/snowball(user.loc)
+	var/datum/reagents/R = source_turf.active_liquid?.group?.reagents
+	if(R)
+		S.color = R.get_average_color().to_rgb()
+	else
```

Comment:
maybe let's not let people piss on the floor and make yellow snowballs...

Reply:
But... but.... okay.

Reply:
People.  People are why we can't have gross things.

Reply:
~~Maybe let them do it, but if they do it more than once a rabid husky jumps from the snowpile and rips them limb from limb.~~

Removing.